### ISSUE TITLE
CSS disappears when importing from external module

### ISSUE THREAD

**Issue by myrsmedenstorytel** [Elderjs/elderjs#86]
Hello! I am trying to update to ElderJS version 1.1.7, from previously using 1.0.6 and I have some issues when using an external library with Svelte components.

I have cloned the template repo and imported an external component in the file Home.svelte and that works well.
However, if I include something from the same library in Simple.svelte (the same or another component), then we lose all CSS in both the Home route and the Simple route that should come from the external library. 

If I have a look at the compiled JS for the route, I can see that the `module.exports._cssIncluded` is reduced to include only the local component files, whereas it was a long list of components and sub-components when it was working. 

So in short: Importing components from an external library in multiple files causes all external CSS to disappear, however importing components in one file and one file only works.

**Comment by nickreese** [comment]
@myrsmedenstorytel Can you push that to a repo I can test on?

**Comment by myrsmedenstorytel** [comment]
Ah yes!
Here is a repository which is a clone of the template, but that includes a button from an external library:
https://github.com/myrsmedenstorytel/elderjs-external-component-css
You can see the button on the home page and in the simple route. If you remove the import (and the usage) in one of the route files Home.svelte or Simple.svelte, then the button will gain its CSS. 
And here is the external lib that just exports a button: https://github.com/myrsmedenstorytel/test-external-svelte-library

Let me know if there are any issues or if you get another behaviour!

**Comment by nickreese** [comment]
Fixed in the latest release. 

The solution is a bit hack-y due to the way rollup doesn't tell us the svelte file imported... instead it just tells us the name of the named import used. 

So for button, we're literally looking for any file that was imported named "Button.svelte"

Unfortunately there isn't a better solution that I see. Just watch out for naming conflicts. 

Relevant code if you can see a way to improve it.

https://github.com/Elderjs/elderjs/commit/a18a6bac92a313716f6a95c9679923c11c0c8c6b

**Comment by myrsmedenstorytel** [comment]
Thanks a lot, @nickreese!
It works well for this particular case, however, if the Button component imports another svelte component, that CSS will not be resolved. Again, it works if we only import the button once but not if we have imports in multiple files. 

I have updated the example repository to contain such an example with nested components.

**Comment by nickreese** [comment]
@myrsmedenstorytel Unfortunately I don't see a solution with the current implementation due to what Rollup gives us. 

Here is a the `chunk` for the `Home.svelte`. 

```javascript
{
  exports: [ 'default' ],
  facadeModuleId: '/Users/nick/repos/elderjs/temp/elderjs-external-component-css/src/routes/home/Home.svelte',
  isDynamicEntry: false,
  isEntry: true,
  isImplicitEntry: false,
  modules: [Object: null prototype] {
    '/Users/nick/repos/elderjs/temp/elderjs-external-component-css/src/routes/home/Home.css': {
      originalLength: 13073,
      removedExports: [],
      renderedExports: [],
      renderedLength: 0
    },
    '/Users/nick/repos/elderjs/temp/elderjs-external-component-css/src/routes/home/Home.svelte': {
      originalLength: 5788,
      removedExports: [],
      renderedExports: [Array],
      renderedLength: 15889
    }
  },
  name: [Getter],
  type: 'chunk',
  code: undefined,
  dynamicImports: [],
  fileName: 'routes/Home.js',
  implicitlyLoadedBefore: [],
  importedBindings: {
    'index-f52204f9.js': [
      'create_ssr_component',
      'escape',
      'each',
      'validate_component',
      'add_attribute'
    ],
    'components/HookDetail.js': [ 'default' ],
    'components/BlogTeaser.js': [ 'default' ],
    'components/Clock.js': [],
    'index-44574cce.js': [ 'Button' ]
  },
  imports: [
    'index-f52204f9.js',
    'components/HookDetail.js',
    'components/BlogTeaser.js',
    'components/Clock.js',
    'index-44574cce.js'
  ],
  map: undefined,
  referencedFiles: []
}
```

A `plugins/components/Icon.css` is emitted by the Svelte rollup plugin. 

```
plugins/components/Icon.css .icon.svelte-wym070{background-color:#fff;border-radius:10px;width:10px;height:10px;color:#000}
/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiSWNvbi5zdmVsdGUiLCJzb3VyY2VzIjpbIkljb24uc3ZlbHRlIl0sInNvdXJjZXNDb250ZW50IjpbIjxzdHlsZT5cbiAgLmljb24ge1xuICAgIGJhY2tncm91bmQtY29sb3I6ICNmZmY7XG4gICAgYm9yZGVyLXJhZGl1czogMTBweDtcbiAgICB3aWR0aDogMTBweDtcbiAgICBoZWlnaHQ6IDEwcHg7XG4gICAgY29sb3I6ICMwMDA7XG4gIH1cbi8qIyBzb3VyY2VNYXBwaW5nVVJMPWRhdGE6YXBwbGljYXRpb24vanNvbjtiYXNlNjQsZXlKMlpYSnphVzl1SWpvekxDSnpiM1Z5WTJWeklqcGJJbTV2WkdWZmJXOWtkV3hsY3k5MFpYTjBMV1Y0ZEdWeWJtRnNMWE4yWld4MFpTMXNhV0p5WVhKNUwzTnlZeTlqYjIxd2IyNWxiblJ6TDI1dlpHVmZiVzlrZFd4bGN5OTBaWE4wTFdWNGRHVnlibUZzTFhOMlpXeDBaUzFzYVdKeVlYSjVMM055WXk5amIyMXdiMjVsYm5SekwwbGpiMjR1YzNabGJIUmxJbDBzSW01aGJXVnpJanBiWFN3aWJXRndjR2x1WjNNaU9pSTdSVUZEUlR0SlFVTkZMSE5DUVVGelFqdEpRVU4wUWl4dFFrRkJiVUk3U1VGRGJrSXNWMEZCVnp0SlFVTllMRmxCUVZrN1NVRkRXaXhYUVVGWE8wVkJRMklpTENKbWFXeGxJam9pYm05a1pWOXRiMlIxYkdWekwzUmxjM1F0WlhoMFpYSnVZV3d0YzNabGJIUmxMV3hwWW5KaGNua3ZjM0pqTDJOdmJYQnZibVZ1ZEhNdlNXTnZiaTV6ZG1Wc2RHVWlMQ0p6YjNWeVkyVnpRMjl1ZEdWdWRDSTZXeUpjYmlBZ0xtbGpiMjRnZTF4dUlDQWdJR0poWTJ0bmNtOTFibVF0WTI5c2IzSTZJQ05tWm1ZN1hHNGdJQ0FnWW05eVpHVnlMWEpoWkdsMWN6b2dNVEJ3ZUR0Y2JpQWdJQ0IzYVdSMGFEb2dNVEJ3ZUR0Y2JpQWdJQ0JvWldsbmFIUTZJREV3Y0hnN1hHNGdJQ0FnWTI5c2IzSTZJQ013TURBN1hHNGdJSDFjYmlKZGZRPT0gKi88L3N0eWxlPlxuXG48ZGl2IGNsYXNzPVwiaWNvblwiPio8L2Rpdj5cbiJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFDRSxLQUFLLGNBQUMsQ0FBQyxBQUNMLGdCQUFnQixDQUFFLElBQUksQ0FDdEIsYUFBYSxDQUFFLElBQUksQ0FDbkIsS0FBSyxDQUFFLElBQUksQ0FDWCxNQUFNLENBQUUsSUFBSSxDQUNaLEtBQUssQ0FBRSxJQUFJLEFBQ2IsQ0FBQyJ9 */
```

I just don't see a way to tie the two together.

**Comment by nickreese** [comment]
The best bet is to probably allow users to write all of the CSS to a location. This is easy compared to the current implementation and may be released in the future when I have time to get to it or someone can assist.

**Comment by myrsmedenstorytel** [comment]
Okay! Yes, then it sounds like it might be trickier to traverse the complete tree in order to get the correct CSS for all nested components.

We have been considering to try to get all CSS into one single file and then to a critical calculation to just inline above-the-fold-CSS, so for us that sounds like a very good solution. 

Thank you for investigating!

**Comment by nickreese** [comment]
@myrsmedenstorytel In discord I recently read that that the Svelte Rollup plugin may change its implementation and stop emitting CSS all together. In the future they will likely encourage users to use a different plugin.

That said:
1. I am open to PRs to fix this specific issue of imports within imports. The rollup expert I lean on says this is doable, but it will be a major undertaking. I don't have a business case for it and neither do my investments so I'm not going to jump down that rabbit hole without community support. 
1. As far as writing to a specific file, this isn't hard to pull off, but I am reluctant to expand the CSS functionality that Elder.js offers until it is clearer what the broader ecosystem Elder.js depends on is doing. If the svelte rollup plugin stops emitting CSS in the future whatever implementation is done today may be rendered obsolete in the future causing breaking changes outside of my control.

If you decide to dive into this PR please ping me on the Svelte Discord.

**Comment by nickreese** [comment]
@myrsmedenstorytel please look at #100.
